feat: water body api (LAN-764)

barredterra opened the pull request:
- feat: water body api (LAN-764)
- docs: organization and water body API (LAN-764)
- Import water bodies from GeoJSON file

Todo:

- [x] add managing organization for each water body to the API response
- [x] ensure map is read-only for role "LANDA Member"

barredterra committed:
feat: water body api (LAN-764)

barredterra committed:
docs: organization and water body API (LAN-764)

barredterra committed:
fix: specify allowed methods

barredterra committed:
feat: add geolocation to water body (LAN-764)

barredterra committed:
feat: add field Display in Fishing Guide (LAN-764)

barredterra committed:
Merge branch 'version-13-hotfix' into water-body-api

barredterra committed:
feat: add command for importing geojson (LAN-764)

barredterra committed:
feat: add field "Guest Passes Available" (LAN-764)

barredterra committed:
feat: expire current information (LAN-764)

barredterra committed:
chore: disable semgrep

barredterra committed:
Merge branch 'version-13-hotfix' into water-body-api

marination commented on docs/cli.md:
```diff
@@ -0,0 +1,42 @@
+### Create Demo Accounts
+
+This app provide a bench command `make-demo-accounts`. This is useful for testing permissions. You can use it to create a **Member**, **Member Function** and **User** for every existing **Member Function Category**, in an existing local organization.
```
```suggestion
This app provides a bench command `make-demo-accounts`. This is useful for testing permissions. You can use it to create a **Member**, **Member Function** and **User** for every existing **Member Function Category**, in an existing local organization.
```

marination commented on docs/cli.md:
```diff
@@ -0,0 +1,42 @@
+
+### Import water bodies from GeoJSON
+
+This app provide a bench command `import-geojson`. This is useful for importing water body shapes from a GeoJSON file.
```
```suggestion
This app provides a bench command `import-geojson`. This is useful for importing water body shapes from a GeoJSON file.
```

marination commented on landa/commands/__init__.py:
```diff
@@ -95,6 +98,50 @@ def update_organization_series(context, dry_run=False):
+			continue
+
+		feature["properties"].pop("GEW_ID")
+		feature["properties"].pop("GEW_NA")
```
Import breaks if there is no `GEW_NA` key. Is this key mandatory like the `GEW_ID` key. If so it must be mentioned in the docs or else handled.
```py
 File ".../apps/landa/landa/commands/__init__.py", line 132, in import_features
    feature["properties"].pop("GEW_NA")
KeyError: 'GEW_NA'
```

marination commented:
Tested the rest. LGTM after changes are resolved.

<img width="1319" alt="Screenshot 2023-06-06 at 2 26 40 PM" src="https://github.com/alyf-de/landa/assets/25857446/face0ee7-4de7-4377-be58-99b7aea9f9ce">

_imported from geojson file_

barredterra committed:
docs: fix typo

Co-authored-by: Marica <maricadsouza221197@gmail.com>

barredterra committed:
fix: don't fail on missing keys (LAN-764)

marination commented:
@barredterra good to merge?

barredterra commented:
@marination there is still an outstanding feature (custom markers) that was supposed to be part of this PR. Let's keep this open for now. I'll check with the customer if we should merge this as a first step.

barredterra committed:
Merge remote-tracking branch 'upstream/version-13-hotfix' into water-body-api

barredterra committed:
fix: set permlevel for water body map (LAN-764)

barredterra committed:
feat: return fish species as simple list

barredterra committed:
refactor: save one variable

barredterra committed:
feat: add organizations to API (LAN-764)

barredterra committed:
fix: water body permissions

Removed system manager and added LANDA Regional Water Body Management
for permlevel 1

barredterra committed:
fix: Guest Passes Available by default

barredterra commented:
:tada: This PR is included in version 0.9.0 :tada:

The release is available on [GitHub release](https://github.com/alyf-de/landa/releases/tag/v0.9.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
